crates/isograph_schema/src/validate_schema.rs

```diff
@@ -1055,13 +1071,20 @@ fn validate_field_type_exists_and_is_linked(
                     }
                 }
             }
-            FieldType::ClientField(_) => Err(WithLocation::new(
-                ValidateSelectionsError::FieldSelectedAsLinkedButTypeIsClientField {
-                    field_parent_type_name: field_parent_object.name,
-                    field_name: linked_field_name,
-                },
-                linked_field_selection.name.location,
-            )),
+            FieldType::ClientField(client_field_id) => {
+                let client_field = &client_fields[client_field_id.as_usize()];
+
+                match client_field.variant {
```

Comment:
1. Here given that a selected field is a client pointer we need to validate it's selections

to do it, we have to thread `client_fields` and `client_pointers` through all the functions